sw/device/silicon_creator/manuf/base/personalization_data.h

```diff
@@ -0,0 +1,80 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_MANUF_BASE_PERSONALIZATION_DATA_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_MANUF_BASE_PERSONALIZATION_DATA_H_
+#include <stdint.h>
+
+/**
+ * Personalization data is sent between the device and the host during the
+ * device provision. Personalization data is laid out as a sequence of
+ * concatenated LTV objects. The length and type field of the objects are packed
+ * in a 16 bit integer:
+ *  d15                                           d0
+ *  +-------------+--------------------------------+
+ *  | 4 bit type  |   12 bits total  object size   |
+ *  +-------------+--------------------------------+
+ *
+ * The following object types have been defined so far:
+ */
+typedef enum perso_object_type {
+  kPersoObjectTypeTpmNotEndorsed = 0,
+  kPersoObjectTypeTpmEndorsed = 1,
+  kPersoObjectTypeDevSeed = 2,
+} perso_object_type_t;
+
+typedef uint16_t object_header_t;
+typedef uint16_t cert_header_t;
+typedef uint16_t dev_seed_header_t;
+
+typedef enum obj_header_fields {
+  kObjhSizeFieldShift = 0,
+  kObjhSizeFieldWidth = 12,
+  kObjhSizeFieldMask = (1 << kObjhSizeFieldWidth) - 1,
+  kObjhTypeFieldShift = kObjhSizeFieldWidth,
+  kObjhTypeFieldWidth = sizeof(object_header_t) * 8 - kObjhSizeFieldWidth,
+  kObjhTypeFieldMask = (1 << kObjhTypeFieldWidth) - 1,
+} obj_header_fields_t;
+
+/**
+ * The x509 certificate is prepeneded by a 16 bits header followed by the ASCII
+ * characters of the certificate name, followed by the certificate body.
+ *
+ * The certificate header includes 4 bits for the certificate name length then
+ * the full size of the certificate object (header size + name lenght +
+ * cetificate size).
+ *
+ *  d15                                           d0
+ *  +-------------+--------------------------------+
+ *  | 4 bit length|       12 bits total size       |
+ *  +-------------+--------------------------------+
+ */
+typedef enum cert_header_fields {
+  kCrthSizeFieldShift = 0,
+  kCrthSizeFieldWidth = 12,
+  kCrthSizeFieldMask = (1 << kCrthSizeFieldWidth) - 1,
```

Comment:
separate each field block with a comment of what it is. (same above for generic object)

Also, can we make this more generic, i.e., add another field to the header that includes the cert type (X509 or CWT). In the future, we would like to be able to support other cert formats besides X509 (i.e., CWT certs encoded in CBOR format)